### PR TITLE
read audio in memory

### LINKED ISSUE #38: Prediction fails on Red Hat OpenShift
The following error is raised, when a prediction is performed:
```

[2019-08-05 21:56:03,078] ERROR in app: Exception on /model/predict [POST]
--
  | Traceback (most recent call last):
  | File "/opt/conda/lib/python3.6/site-packages/flask/app.py", line 1832, in full_dispatch_request
  | rv = self.dispatch_request()
  | File "/opt/conda/lib/python3.6/site-packages/flask/app.py", line 1818, in dispatch_request
  | return self.view_functions[rule.endpoint](**req.view_args)
  | File "/opt/conda/lib/python3.6/site-packages/flask_restplus/api.py", line 319, in wrapper
  | resp = resource(*args, **kwargs)
  | File "/opt/conda/lib/python3.6/site-packages/flask/views.py", line 88, in view
  | return self.dispatch_request(*args, **kwargs)
  | File "/opt/conda/lib/python3.6/site-packages/flask_restplus/resource.py", line 44, in dispatch_request
  | resp = meth(*args, **kwargs)
  | File "/opt/conda/lib/python3.6/site-packages/flask_restplus/marshalling.py", line 136, in wrapper
  | resp = f(*args, **kwargs)
  | File "/workspace/api/predict.py", line 63, in post
  | file = open("audio.wav", "wb")
  | PermissionError: [Errno 13] Permission denied: 'audio.wav'

```

The issue is that the code assumes that it has write access in the current working directory https://github.com/IBM/MAX-Audio-Classifier/blob/master/api/predict.py#L59-L89. The temporary audio file should be created in a temporary location that any user should have write access to.

### PR DESCRIPTION
**Pull request by bdwyer2**
resolves #38

### CODE REVIEW

File: api/predict.py [hunk 1]
```diff
@@ -55,22 +54,9 @@ def post(self):
...
-        if os.path.exists("audio.wav"):
-            os.remove("audio.wav")
-
-        if '.wav' in str(args['audio']):
-            file = open("audio.wav", "wb")
-            file.write(audio_data)
-            file.close()
-        else:
-            e = BadRequest()
-            e.data = {'status': 'error', 'message': 'Invalid file type/extension'}
```
Comment (kmh4321) [inline api/predict.py]: How do we handle exception for invalid file types?

Comment (bdwyer2) [reply api/predict.py]: I can add the exception back

File: api/predict.py [hunk 2]
```diff
@@ -53,24 +52,17 @@ def post(self):
...
-        # clean up from earlier runs
-        if os.path.exists("audio.wav"):
-            os.remove("audio.wav")
 
-        if '.wav' in str(args['audio']):
-            file = open("audio.wav", "wb")
-            file.write(audio_data)
-            file.close()
-        else:
+        if '.wav' not in str(args['audio']):
```
Comment (xuhdev) [inline api/predict.py]: [python-magic](https://github.com/ahupp/python-magic) or [mimetypes](https://docs.python.org/3.7/library/mimetypes.html) might be more accurate.

Comment (kmh4321) [reply api/predict.py]: Would like to reduce dependencies on external packages. For mimetype I see this [issue](https://stackoverflow.com/questions/44891157/why-wav-format-doesnt-have-same-mimetype-in-different-browsers).

File: tests/test.py [hunk 3]
```diff
@@ -153,5 +153,21 @@ def test_multi_filter():
     assert response['predictions'][1]['probability'] > 0.09
 
 
+def test_invalid_extension():
```
Comment (xuhdev) [inline tests/test.py]: test_invalid_mimetype

File: tests/test.py [hunk 4]
```diff
@@ -153,5 +153,21 @@ def test_multi_filter():
...
+    with open(file_path, 'rb') as file:
+        file_form = {'audio': (file_path, file, 'text/x-python')}
+        r = requests.post(url=model_endpoint, files=file_form)
+
+    assert r.status_code == 400
+
+    response = r.json()
+
+    assert response['status'] == 'error'
+    assert response['message'] == 'Invalid file type/extension'
```
Comment (xuhdev) [inline tests/test.py]: 'Invalid mimetype'

Comment (bdwyer2) [reply tests/test.py]: discussed this offline

**Pull request merged**